src/mongoc/mongoc-cluster.c

```diff
@@ -207,6 +207,32 @@ _mongoc_cluster_run_command (mongoc_cluster_t      *cluster,
 /*
  *--------------------------------------------------------------------------
  *
+ * _mongoc_cluster_is_single_threaded --
```

Comment:
Let's ditch this.  In single threaded you don't need the lock.  In multi-threaded it doesn't matter, because you set that member variable before thread starts, which synchronizes (and it never changes during the life of the object).